### DISCUSSION TITLE
MODULE NOT FOUND, when not at home dir

### DISCUSSION THREAD

**Discussion by tada123** [nodejs/help#4986] [Q&A]
### Version

v18.9.0

### Platform

Linux hostname-pc 5.19.2-zen1-1-zen #1 ZEN SMP PREEMPT_DYNAMIC Wed, 17 Aug 2022 14:28:00 +0000 x86_64 GNU/Linux

### What steps will reproduce the bug?

1: cd to directory another than $HOME
2: Just install module, do `npm link $module` in home folder, check, that folder is created or linked.
3: try to run `require('module')`

### How often does it reproduce? Is there a required condition?

sometimes yes, sometimes not, (now ALWAYS)

### What is the expected behavior?

That `require` will work, when ~/node_modules contains module with exact name and package.json with all the needed stuff

### What do you see instead?

> require('mail-confirm')
Uncaught Error: Cannot find module 'mail-confirm'
Require stack:
- <repl>
    at Module._resolveFilename (node:internal/modules/cjs/loader:955:15)
    at Module._load (node:internal/modules/cjs/loader:803:27)
    at Module.require (node:internal/modules/cjs/loader:1021:19)
    at require (node:internal/modules/cjs/helpers:103:18) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '<repl>' ]
}

### Additional information

some modules worked, some not, but now nothing

**Comment by bnoordhuis** [comment]
I've converted this to a discussion because this seems like a misunderstanding of how npm and the module loader work, not a bug in node.

The module loader is described in node's documentation, npm's docs you can find here: https://docs.npmjs.com/

**Reply by bf** [reply]
I actually have the same problem. You import a file from a lib/ folder and within that module it is unable to import 'pg'
